test/test_dataarray.py

```diff
@@ -111,3 +113,29 @@ def test_geodataframe_caching(self):
 
         # override will recompute the grid
         assert gdf_start is not gdf_end
+
+    def test_nodal_average(self):
+
+        # test on a node-centered dataset
+        uxds = ux.open_dataset(gridfile_geoflow, dsfile_v1_geoflow)
+
+        v1_nodal_average = uxds['v1'].nodal_average()
+
+        # final dimension should match number of faces
+        assert v1_nodal_average.shape[-1] == uxds.uxgrid.n_face
```

Comment:
It's probably better to use `self.AssertEqual()` to cohere with other test cases